src/cargo/ops/cargo_clean.rs

```diff
@@ -21,17 +21,24 @@ pub fn clean(manifest_path: &Path, opts: &CleanOptions) -> CargoResult<()> {
 
     // If we have a spec, then we need to delete some packages, otherwise, just
     // remove the whole target directory and be done with it!
+    //
+    // Note that we don't bother grabbing a lock here as we're just going to
+    // blow it all away anyway.
```

Comment:
That's still a race. Doesn't this mean that if you run `cargo clean` while you are building that they'll clobber each other? Is there a technical reason it's hard to lock this?